test/Microsoft.AspNetCore.Proxy.Test/WebSocketsTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Proxy.Test
+{
+    public class WebSocketsTest
+    {
+        private static async Task<string> ReceiveTextMessage(WebSocket socket, int maxLen = 4096)
+        {
+            var recvBuff = new byte[maxLen];
+            var received = 0;
+            while (true)
+            {
+                var res = await socket.ReceiveAsync(new ArraySegment<byte>(recvBuff, received, maxLen - received), CancellationToken.None);
+                Assert.Equal(WebSocketMessageType.Text, res.MessageType);
+                received += res.Count;
+                if (res.EndOfMessage)
```

Comment:
nit: To keep consistent with our coding style can you wrap the body of the if in curly braces